BAU: Use the new metadata published by the apps

CharlesIC opened the pull request:
- Changed entity IDs to match the new metadata endpoints
  - Rename secrets in K8s cluster to be more clear by better describing what purpose they serve
  - Add a separate metadata signing cert for Stub Connector

CharlesIC committed:
BAU: Use the new metadata published by the apps

  - Changed entity IDs to match the new metadata endpoints
  - Rename secrets in K8s cluster to be more clear by better describing what purpose they serve
  - Add a separate metadata signing cert for Stub Connector

Author: @CharlesIC

blairboy362 commented on chart/templates/metadata-ingress-virtual-service.yaml:
```diff
@@ -16,6 +16,6 @@ spec:
     - route:
       - destination:
-          host: {{ .Release.Name }}-proxy-metadata
+          host: {{ .Release.Name }}-proxy-node-metadata
```
This is pointing at the service that is created by the VMC when processing a `Metadata` resource, which might not be what we want. Potentially lengthy discussion to have around this that I'm not going to attempt via github comments.

CharlesIC replied:
Actually we don't need this anymore - that's the thing I was talking about this morning. Let's discuss tomorrow.

CharlesIC committed:
BAU: Change istio config to allow the Stub Connector to access the new Proxy Node metadata

Istio rules need to be adjusted so that Stub Connector can access the Proxy Node metadata published by Gateway

Author: @CharlesIC

blairboy362 commented on chart/templates/_helpers.tpl:
```diff
@@ -5,7 +5,7 @@
 
 {{- define "connector.host" -}}
```
Thinking about it...this block can be simplified further. We no longer need the `if...else`; just the `printf` inside the `if` part.

blairboy362 commented on chart/templates/_helpers.tpl:
```diff
@@ -5,7 +5,7 @@
 
 {{- define "connector.host" -}}
 {{- if .Values.stubConnector.enabled -}}
```
```suggestion
```

blairboy362 commented on chart/templates/_helpers.tpl:
```diff
@@ -5,7 +5,7 @@
 {{- if .Values.stubConnector.enabled -}}
-{{- printf "%s-%s.%s" .Release.Name "connector" (required "global.cluster.domain is required" .Values.global.cluster.domain) | trimSuffix "-" -}}
+{{- printf "%s" (include "connector.metadata.host" .) | trimSuffix "-" -}}
 {{- else -}}
```
```suggestion
```

blairboy362 commented on chart/templates/_helpers.tpl:
```diff
@@ -5,7 +5,7 @@
-{{- printf "%s-%s.%s" .Release.Name "connector" (required "global.cluster.domain is required" .Values.global.cluster.domain) | trimSuffix "-" -}}
+{{- printf "%s" (include "connector.metadata.host" .) | trimSuffix "-" -}}
 {{- else -}}
 {{- printf "%s" .Values.connector.host -}}
```
```suggestion
```

blairboy362 commented on chart/templates/_helpers.tpl:
```diff
@@ -5,7 +5,7 @@
+{{- printf "%s" (include "connector.metadata.host" .) | trimSuffix "-" -}}
 {{- else -}}
 {{- printf "%s" .Values.connector.host -}}
 {{- end -}}
```
```suggestion
```

blairboy362 commented:
You may also be able to remove "metadata.host" from the `_helpers.tpl` (confirm it with a search, but I hope it is unused now).
